### ISSUE TITLE
process.stdin.drain() hangs randomly when writing reasonably big buffers

### ISSUE THREAD

**Issue by seirl** [python/asyncio#381]
This code hangs at the `stdin.drain()` line for some reasonably big buffers. Here's a minimal example:

```python
#!/usr/bin/env python3

import asyncio
import time

async def drain_time(length):
    proc = await asyncio.create_subprocess_exec('/bin/cat',
            stdin=asyncio.subprocess.PIPE,
            stdout=asyncio.subprocess.PIPE,
    )

    data = b'0' * length
    proc.stdin.write(data)
    t0 = time.time()
    await proc.stdin.drain()
    r = time.time() - t0
    proc.stdin.close()

    await proc.wait()
    return r

async def main():
    for i in range(1, 9):
        for j in range(1, 10):
            print(10 ** i * j, await drain_time(10**i * j))

asyncio.get_event_loop().run_until_complete(main())
```
It starts outputting lines like:

```
10 7.152557373046875e-06                     
20 5.245208740234375e-06
30 5.4836273193359375e-06
```

and then hangs either at 200000 or 300000 bytes.

Attached is the python trace from the start of the program until it hangs: [trace.zip](https://github.com/python/asyncio/files/376703/trace.zip)

One thing I noted is that the first fail happens when at `asyncio/streams.py:209`, `self._paused` is True (it's False for all the other iterations).

You can see at the end of the attached trace that the hanging line is:

`selectors.py(441):                 fd_event_list = self._epoll.poll(timeout, max_ev)`

**Comment by seirl** [comment]
It also happens with uvloop, but after 400000 bytes or 500000 bytes.

**Comment by seirl** [comment]
uname -a : `Linux hostname 4.4.0-21-generic #37-Ubuntu SMP Mon Apr 18 18:33:37 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux`

Python version: ```Python 3.5.1+ (default, Mar 30 2016, 22:46:26) 
[GCC 5.3.1 20160330] on linux```

**Comment by arthurdarcet** [comment]
Also, even with `stdout=DEVNULL`, the draining time grows exponentially

```
1000000 0.018378019332885742
2000000 0.041895151138305664
3000000 0.18218016624450684
4000000 0.3633248805999756
5000000 0.5759620666503906
6000000 0.8171072006225586
7000000 1.0535521507263184
8000000 1.451490879058838
9000000 1.8422210216522217
10000000 2.216843843460083
20000000 9.008002042770386
30000000 19.805752992630005
40000000 45.31144189834595
50000000 60.9793598651886
60000000 95.35793805122375
70000000 123.93864011764526
80000000 169.90427994728088
```

And I might be mistaken, but I feel like 2.2 seconds to write 8 MB of data in a cat process is way too slow

**Comment by seirl** [comment]
Oh, you're right, it seems like passing stdout=DEVNULL fixes the hanging issue.

**Comment by seirl** [comment]
It looks solved in python 3.5.2, I'm closing the issue... :)
